index.js

```diff
@@ -219,6 +236,9 @@ URL.prototype.set = function set(part, value, fn) {
       url.hostname = value;
       url.port = '';
     }
+
+    if (isIPv6(url.hostname)) url.hostname = url.hostname.slice(1, -1);
```

Comment:
It could use a comment that ipv6 uses brackets and that we want to remove them from the host name.